lib/services/users.pm

```diff
@@ -134,6 +134,8 @@ sub restore_passwd {
 
 # remove test user
 sub remove_test_user {
+    # We need wait a bit more time for the user related process to quit even after logout the user.
+    assert_script_run('for ((i=5; i>0; i--)) do if (! ps -e -u test > /dev/null); then break; else sleep 3; continue; fi done');
     assert_script_run("userdel -r $newUser");
```

Comment:
May be "userdel -rf $testuser" can work well too.

Reply:
For us, on x11, we need wait the process quit while not force kill it, else it may cause unexpected failure such as console issue.